tests/test_base_runner.py

```diff
@@ -0,0 +1,74 @@
+#
+# Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from datetime import datetime
+from unittest.mock import patch
+
+import pytest
+from cloudai import BaseRunner
+
+
+@pytest.fixture
+def mock_datetime_now():
+    with patch("cloudai._core.base_runner.datetime") as mock_datetime:
+        mock_datetime.now.return_value = datetime(2024, 1, 1, 12, 0, 0)
+        mock_datetime.strftime = datetime.strftime
+        yield mock_datetime
+
+
+def test_setup_output_directory(mock_datetime_now, tmp_path):  # noqa
```

Comment:
Can we avoid using `# noqa`? It can fight back at some point but won't letting us know about real issues.

Reply:
Linters are complaining that mock_datetime_now is not being used. How can I fix these errors?

Reply:
```
$ vulture . | grep 100       
tests/test_base_runner.py:32: unused variable 'mock_datetime_now' (100% confidence)
tests/test_base_runner.py:47: unused variable 'mock_datetime_now' (100% confidence)
```

Reply:
Ah, I see. Guess it should be a feature request for `vulture`, they don't handle `pytest`'s magic well.

Can we disable only `vulture` only for these lines? For `ruff` we can do: `... # noqa: <rule id>`.

Reply:
Updated.